hollow/src/main/java/com/netflix/hollow/core/memory/HollowUnsafeHandle.java

```diff
@@ -38,10 +44,102 @@
             log.log(Level.SEVERE, "Unsafe access failed", e);
         }
         unsafe = u;
+        singleton = new HollowUnsafeHandle();
+    }
+
+    public static HollowUnsafeHandle getUnsafe() {
+        return singleton;
+    }
+
+    public void putOrderedLong(Object o, long l, long l1) {
+        if (UNALIGNED_ALLOWED) {
+            unsafe.putOrderedLong(o, l, l1);
+        } else {
+            // Assume little endian
+            unsafe.putByteVolatile(o, l, (byte) (l1));
+            unsafe.putByteVolatile(o, l + 1, (byte) (l1 >>> 8));
+            unsafe.putByteVolatile(o, l + 2, (byte) (l1 >>> 16));
+            unsafe.putByteVolatile(o, l + 3, (byte) (l1 >>> 24));
+            unsafe.putByteVolatile(o, l + 4, (byte) (l1 >>> 32));
+            unsafe.putByteVolatile(o, l + 5, (byte) (l1 >>> 40));
+            unsafe.putByteVolatile(o, l + 6, (byte) (l1 >>> 48));
+            unsafe.putByteVolatile(o, l + 7, (byte) (l1 >>> 56));
+        }
+    }
+
+    public long getLong(Object o, long l) {
+        if (UNALIGNED_ALLOWED) {
```

Comment:
I'm curious, if you change this to `if (true)` do you see the SIGBUS? In our testing the only place where we saw SIGBUS was when the delta application code calls `putOrderedLog`. Once we disabled deltas, everything else (initializing from a snapshot, creating indices, un aligned reads, etc.) seemed to work.

Reply:
Asking because of performance criticality here

Reply:
I'm fairly certain that, since the guard is a `final`, the JVM will optimize out the branch not-taken when it compiles this code to native, but I can swap to using two implementations to be more certain.

Reply:
I'll test this and get back to you.